### PR TITLE
Move checkProxy script under WMCore; script to renew proxy

### LINKED ISSUE #9641: Proxy renewal failing in the CERN agents
**Impact of the bug**
WMAgents

**Describe the bug**
We noticed a couple of CERN agents started failing to renew my proxy - which is done through a cron entry - and over the last few days, there has been a couple more agents failing as well. Debugging shows a problem with java, likely related to some sort of update that hit those nodes in the last days..

**How to reproduce it**
Running the command manually works fine, either with or without the agent environment (!)

**Expected behavior**
The proxy renewal should go through. Ideally it would be better to have a component taking care of it, but we can simply fix it for now and keep the same renewal model.

**Additional context and error message**
After piping the error to a log, here is the problem:
```
A credential has been received for user amaltaro in /data/certs/mynewproxy.pem.
Error occurred during initialization of VM
java.lang.Error: Properties init: Could not determine current working directory.
        at java.lang.System.initProperties(Native Method)
        at java.lang.System.initializeSystemClass(System.java:1168)
```

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #9641 

#### Status
not-tested

#### Description
Changes are:
* moved the checkProxy script from my repo over to WMCore (used to check the proxy validity and notify if below a given timeleft)
* converted the cron entry to renew my proxy into a bash script (using the agent's libraries by the way)
* updated the deployment script accordingly

#### Is it backward compatible (if not, which system it affects?)
yes

#### Related PRs
none

#### External dependencies / deployment changes
<Does it require deployment changes? Does it rely on third-party libraries?>

### CODE REVIEW

File: deploy/checkProxy.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            print('Proxy information: {}'.format(proxyInfo))
+        timeLeft = []
+        for line in proxyInfo:
+            if line.find('timeleft') > -1:
+                dateReg = re.compile('\d{1,3}[:/]\d{2}[:/]\d{2}')
+                timeLeft = dateReg.findall(line)[0]
+                timeLeft = timeLeft.split(':')[0]
+                continue
+    else:
+        msg = "Valid proxy file in %s not found. " % host
```
Comment (todor-ivanov) [inline deploy/checkProxy.py:128]: I would change this message to:
"No valid proxy was found in %s"

File: deploy/renew_proxy.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+credname=CREDNAME
+source /data/admin/wmagent/env.sh
+source /data/srv/wmagent/current/apps/wmagent/etc/profile.d/init.sh
+myproxy-get-delegation -v -l amaltaro -t 168 -s 'myproxy.cern.ch' -k $credname -n \
+  -o /data/certs/mynewproxy.pem && voms-proxy-init -rfc -voms cms:/cms/Role=production -valid 168:00 \
```
Comment (todor-ivanov) [inline deploy/renew_proxy.sh:6]: In this line we are still using voms-proxy-init through the system paths as they are configured by default. Which in many cases (especially for the CERN machines) points to voms-client-java. [1]

[1]
```
[vocms0252]$ ll `which voms-proxy-init`
lrwxrwxrwx. 1 root root 33 Nov 17  2018 /usr/bin/voms-proxy-init -> /etc/alternatives/voms-proxy-init
[vocms0252]$ ll /etc/alternatives/voms-proxy-init
lrwxrwxrwx. 1 root root 25 Nov 17  2018 /etc/alternatives/voms-proxy-init -> /usr/bin/voms-proxy-init3
[vocms0252]$ cat /usr/bin/voms-proxy-init3
#!/bin/sh
VOMS_CLIENTS_JAVA_OPTIONS=${VOMS_CLIENTS_JAVA_OPTIONS:-"-XX:+UseSerialGC -Xmx16m"}
java $VOMS_CLIENTS_JAVA_OPTIONS -cp $(build-classpath voms-clients-java voms-api-java canl-java bcpkix bcprov commons-cli commons-io) org.italiangrid.voms.clients.VomsProxyInit "$@"
```

**Review by todor-ivanov — APPROVED** [review]
Having it all done the way it is suggested in this PR is much better than the current implementation. It improves the situation a lot. Although, even with a dedicate utilitarian script we still do not avoid the risk of eventual crash of the Java voms-clients with the next system upgrade(due to a silent java version change etc.). Please take a look at the comment I made directly in the issue. This is a problem already faced before, with a proven solution.

File: deploy/checkProxy.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    for line in proxyInfo:
+        message += "%s\n" % line
+    # echo %msg | mail -s 'HOST proxy status' MAIL_ADD
+    command = " echo \"%s\" | " % (message)
+    command += "mail -s '%s: Proxy status'" % (host)
+    command += " %s" % (mail)
+
+    if verbose:
+        print("Running email command: {}".format(command))
+    retCode = os.system(command)
```
[the remark this answers is not in the document]
Comment (amaltaro) [reply deploy/checkProxy.py:106]: Yes, I considered that library too. But then this script becomes dependent on the libraries we ship with WMAgent, which might bring other problems too.
I have just made the necessary changes to use the EmailAlert, but then I started getting an `undefined symbol` problem.

File: deploy/checkProxy.py (hunk 1 above)
Comment (amaltaro) [reply deploy/checkProxy.py:128]: Changing

**Comment by amaltaro** [comment]
Todor, I believe I already tried to change to the c++ version of the voms-client in the past, and failed.
In this PR, I also tried to use the EmailAlert library, but then we hit other problems that are likely related to this issue:
https://github.com/dmwm/WMCore/issues/7936

So, I'm keeping it as it is for now, I don't think it's worth the effort at the moment. Feel free to create another issue (and the final fix), if you know how exactly this can be improved.
Thanks for your comments though.

**Comment by todor-ivanov** [comment]
Thanks @amaltaro for letting me know about your past effort. That was good to read. Yes, it is logical that with the C clients we introduce dependency on library versions from the comp environment. And I can see the other eventual version misalignment may obviously come from yet another package that lives outside comp - HTCondor. We of course can spare the effort of just moving the pain from one place to another and we may continue living with the java version.

**Review by todor-ivanov — COMMENTED** [review]
looks good to me

**Comment by amaltaro** [comment]
AgentStatusWatcher is the right place to have these checks. I actually don't know why it hasn't been implemented that way yet, likely because there is one of these problems already seen. 
Still, that's the correct way of fixing it!

**Comment by amaltaro** [comment]
Sigh...of course FNAL and CERN nodes have different version of the voms-proxy-init. And now the proxy creation fails with
```
A credential has been received for user amaltaro in /data/certs/mynewproxy.pem.
voms-proxy-init: symbol lookup error: voms-proxy-init: undefined symbol: _ZN8vomsdataC1ESsSs
```
on the FNAL nodes.

**Pull request merged**